src/index.js

```diff
@@ -0,0 +1,130 @@
+	 * @property {(number|null)} id - integer request ID of the message for "Request" protobuf definitions, null otherwise.
+	 * @property {(Function | null)} replyMessage The corresponding reply message to a given "Request" message, null otherwise.
+   */
+
+	/**
+	 * @returns {Array} valid strings that can be passed to getDefinition(). Includes all Request/Reply style messages as well as non request messages and enums.
+	 */
+	static getDefinitions() {
+		const namespaces = this.getNamespaces();
+		const returnThis = [];
+		for (const key of Object.keys(this._pbjsObjects)) {
+			if (namespaces.includes(key)) {
+				const namespace = namespaces[namespaces.indexOf(key)];
+				for (const namespacedKey of Object.keys(this._pbjsObjects[namespace])) {
+					returnThis.push(`${namespace}.${namespacedKey}`);
+				}
+			} else {
+				returnThis.push(key);
+			}
+		}
+		return returnThis;
+	}
+
+	/**
+	 * @returns {Array} valid dot prefixes to getDefinition() arguments (i.e. the "cellular" from "cellular".GetIccidRequest, etc)
+	 */
+	static getNamespaces() {
+		return [
+			'wifi',
+			'logging',
+			'mesh',
+			'cloud',
+			'cellular'
+		];
+	}
+
+	static _getIDFromJSON(protobufMessageName) {
+		let rootJSONObject;
+		if (this._isNamespaced(protobufMessageName)) {
+			const [namespace, nonNamespacedName] = this._getNamespaceAndMessageName(protobufMessageName);
+			rootJSONObject = DeviceOSProtobuf._pbjsJSON.nested.particle.nested.ctrl
+				.nested[namespace].nested[nonNamespacedName];
+		} else {
+			rootJSONObject = DeviceOSProtobuf._pbjsJSON.nested.particle.nested.ctrl
+				.nested[protobufMessageName];
+		}
+
+		try {
+			return rootJSONObject.options['(type_id)'];
+		} catch (e) {
+			throw new Error(`Could not extract request type id from pbjs json where protobufMessageName=${protobufMessageName}`);
+		}
+	}
+
+	static _isNamespaced(protobufMessageName) {
+		return protobufMessageName.includes('.');
+	}
+
+	static _getNamespaceAndMessageName(protobufMessageName) {
+		return protobufMessageName.split('.');
+	}
+
+	// Get's a pbjs generated Function by name or namespace.name
+	static _getProtobufMessage(protobufMessageName) {
+		let rootObject;
+		if (protobufMessageName.includes('.')) {
+			const [namespace, nonNamespacedName] = this._getNamespaceAndMessageName(protobufMessageName);
+			protobufMessageName = nonNamespacedName;
+			rootObject = this._pbjsObjects[namespace];
+		} else {
+			rootObject = this._pbjsObjects;
+		}
+		if (!(protobufMessageName in rootObject)) {
+			throw new Error(`There is no pbjs generated protobuf Function for protobufMessageName=${protobufMessageName}`);
+		}
+		return rootObject[protobufMessageName];
+	}
+
+	static _getProtobufReplyMessage(protobufMessageName) {
+		return this._getProtobufMessage(protobufMessageName.replace('Request','Reply'));
+	}
+}
+
+const pbjsJSONString = fs.readFileSync(`${path.resolve(__dirname)}/pbjs-generated/index.json`);
```

Comment:
Did the following not work?
```suggestion
const pbjsJSONString = require('./pbjs-generated/index.json`);
```
The `fs` module is not available in React Native

Reply:
I'd recommend renaming the file to `pbjs-generated/definitions.json` and not `index.json` so it's not confusing between `index.js` and `index.json`. There's a parameter for `pbjs` for the name of the output file, right?

Reply:
I did try eliminating `fs` and `path` starting with the code change you suggested above. At first, it looked like it didn't work. I just tried again and realized the problem was i was doing that `require` AND yet still running `JSON.parse` on it, aka double parsing. Fixed now.